utils/node_local_test.py

```diff
@@ -1957,7 +1976,40 @@ def test_readdir_hard(self):
         """
         test_dir = join(self.dfuse.dir, 'test_dir')
         os.mkdir(test_dir)
-        count = 30
+        count = 140
+        for idx in range(count):
+            with open(join(test_dir, f'file_{idx}'), 'w'):
+                pass
+
+        files = []
+        files2 = []
+        with os.scandir(test_dir) as entries:
+            with os.scandir(test_dir) as second:
+                files2.append(next(second).name)
+                for entry in entries:
+                    files.append(entry.name)
+                    assert len(files) < count + 2
+                for entry in second:
+                    files2.append(entry.name)
+                    assert len(files2) < count + 2
+
+        print('Reads are from list 2, 1, 1, 2.')
+        print(files)
+        print(files2)
+        assert files == files2
+        assert len(files) == count
```

Comment:
For robustness, should we save the file names from create and verify readdir returns the correct file names?

Reply:
We could but that part is extremely likely to go wrong.

Reply:
But shouldn't it work? If readdir returns the wrong file names, it's functionally not working